library/src/scripts/components/siteNav/SiteNavNode.tsx

```diff
@@ -0,0 +1,240 @@
+            this.setState({
+                open: true,
+            });
+        } else {
+            if (callbackIfAlreadyOpen) {
+                callbackIfAlreadyOpen();
+            }
+        }
+    };
+
+    public close = (callbackIfAlreadyClosed?: any) => {
+        if (this.state.open) {
+            this.setState({
+                open: false,
+            });
+        } else {
+            if (callbackIfAlreadyClosed) {
+                callbackIfAlreadyClosed();
+            }
+        }
+    };
+
+    public openRecursive = () => {
+        if (!this.state.current && this.currentPage()) {
+            this.setState({
+                current: true,
+            });
+            if (this.props.openParent) {
+                this.props.openParent();
+            }
+        }
+    };
+
+    public openSelfAndOpenParent = () => {
+        this.setState({
+            open: true,
+        });
+        if (this.props.openParent) {
+            this.open();
+            this.props.openParent();
+        }
+    };
+
+    public toggle = () => {
+        this.setState({
+            open: !this.state.open,
+        });
+    };
+
+    public handleClick = e => {
+        e.preventDefault();
+        this.toggle();
+    };
+
+    public currentPage(): boolean {
+        if (this.props.location && this.props.location.pathname) {
+            return this.props.location.pathname === this.props.url;
+        } else {
+            return false;
+        }
+    }
+
+    public updateCurrentState() {
+        this.setState({
+            current: this.currentPage(),
+        });
+    }
+
+    public componentDidUpdate(prevProps) {
+        this.openRecursive();
+        if (prevProps.location.pathname !== this.props.location.pathname) {
+            this.updateCurrentState();
+        }
+    }
+
+    public componentDidMount() {
+        this.openRecursive();
+    }
+
+    public render() {
+        const hasChildren = this.props.children && this.props.children.length > 0;
+        const topLevel = this.props.counter === 1;
+        const childrenContents =
+            hasChildren &&
+            this.props.children.map((child, i) => {
+                return (
+                    <SiteNavNode
+                        {...child}
+                        key={"siteNavNode-" + this.props.counter + "-" + i}
+                        counter={this.props.counter! + 1}
+                        openParent={this.openSelfAndOpenParent}
+                        location={this.props.location}
+                        depth={this.props.depth + 1}
+                    />
+                );
+            });
+        const space = `&nbsp;`;
+        return (
+            <li
+                className={classNames("siteNavNode", this.props.className, { isCurrent: this.state.current })}
+                role="treeitem"
+                aria-expanded={this.state.open}
+                style={{ marginLeft: `${(this.props.depth - 1) * 18}px` }}
```

Comment:
I feel like this styling should be done through CSS so we can use our margin variables.

Classes like `isDepth-x` would probably do the trick right?